plutus-use-cases/README.md

```diff
@@ -1,2 +1,257 @@
+                                      contributorOnly &&
+                                      $$(Validation.txSignedBy) p pkCon
+```
+
+The overall result of this branch is the `refundable` value:
+
+```haskell
+                          in refundable
+```
+
+The second branch represents a successful campaign. 
+
+```haskell
+                      Collect -> 
+```
+
+In the `Collect` case, the current blockchain height must be between `deadline` and `collectionDeadline`, the target must have been met, and and transaction has to be signed by the campaign owner.
+
+```haskell
+                          currentHeight > deadline &&
+                          currentHeight <= collectionDeadline &&
+                          totalInputs >= target &&
+                          $$(Validation.txSignedBy) p campaignOwner
+
+              in
+```
+
+Finally, we can return the unit value `()` if `isValid` is true, or fail with an error otherwise.
+
+```haskell
+              if isValid then () else PlutusTx.error ()
+                  ||])
+```
+
+We need to compute the address of a campaign, which amounts to  hashing the output of `mkValidatorScript`:
+
+```haskell
+campaignAddress :: Campaign -> Address'
+campaignAddress cmp = scriptAddress (mkValidatorScript cmp)
+```
+
+Now that we have the validator script, we need to set up wallet actions for contributors and the campaign owner. Contributors put their public key in a data script:
+
+```haskell
+mkDataScript :: PubKey -> DataScript
+mkDataScript pk = DataScript (lifted (Contributor pk))
+```
+
+Wallet actions have the return type `MockWallet ()`, which means that they can use the wallet API to create and submit transactions, query blockchain addresses, and log messages.
+
+```haskell
+contribute :: Campaign -> Value -> MockWallet ()
+contribute cmp amount = do
+```
+
+Contributing to a campaign is easy: We need to pay the value `amount` to a script address, and provide our own public key as the data script.
+
+```haskell
+      pk <- ownPubKey
+      let dataScript = mkDataScript pk
+      tx <- payToScript (campaignAddress cmp) amount dataScript
+```
+
+`tx` is a transaction that pays `amount` to the address of the campaign validator script, using our own public key as the data script. 
+
+```haskell
+      -- TODO: In the original contract we now register a refund trigger for our own contribution, using the hash of `tx`.
+      pure ()
+```
+
+When we want to spend the contributions we need to provide a `RedeemerScript` value. In our case this is just the `CampaignAction`:
+
+```haskell
+mkRedeemer :: CampaignAction -> RedeemerScript
+mkRedeemer action = RedeemerScript (lifted (action))
+```
+
+Collecting the funds:
+
+```haskell
+collect :: Campaign -> MockWallet ()
+collect cmp = 
+      let validatorScript = mkValidatorScript cmp
+          redeemer = mkRedeemer Collect
+      in
+          collectFromScript validatorScript redeemer
+```
```

Comment:
I forgot a bit of text at the end, but beyond that I'm not sure how to end this properly: Should it be a Playground-specific tutorial, or should it explain the whole mockchain? I also want to add a second part that covers blockchain triggers (they are needed for the refund). This won't be covered in the talk but otherwise the contract is incomplete.

Reply:
It's okay for this to be a WIP, but ideally this would take you all the way to running it on the emulator locally (i.e. writing the kind of tests we've written).

Reply:
Added two more sections that will be filled in at a later date.